test/models/report_test.rb

```diff
@@ -3,7 +3,25 @@
 require 'test_helper'
 
 class ReportTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  include Devise::Test::IntegrationHelpers
+
+  test '日報作成者は作成した日報を編集可能であること' do
+    report = reports(:report1)
+    user = users(:user_taro)
+
+    assert report.editable?(user)
+  end
+
+  test '日報作成者以外は日報を編集不可能であること' do
+    report = reports(:report1)
+    user = users(:user_nameless)
+
+    assert_not report.editable?(user)
+  end
+
+  test '日報の作成日をdate型で取得できること' do
+    report = reports(:report1)
+
+    assert_equal Time.zone.today, report.created_on
```

Comment:
```suggestion
    assert_equal Date.current, report.created_on
```

の方が短く書けるので僕はこっちを使います。

あと、「たまたまシステム日付と一致していただけで、実はcreated_atを使ってなかった」という不具合を検知できるように、意図的にシステム日付とずらすとベターかもしれません。

```ruby
report.created_at = "2023-01-15 01:20:30".in_time_zone
assert_equal Date.new(2023, 1, 15), report.created_on
```

Reply:
なるほどです！
システム日付とずらしてテストするよう修正します。